File: dr_quine/Grace/Grace.c
/*
J'aime les poneys
*/
#include <stdio.h>

#define zero 0

#define FCLSE(a) fclose(a)

#define FT(x) int main(){ char eol='\n', qt='"', esc='\\'; char* str="/*%cJ'aime les poneys%c*/%c#include <stdio.h>%c%c#define zero 0%c%c#define FCLSE(a) fclose(a)%c%c#define FT(x) int main(){ char eol='%cn', qt='%c', esc='%c%c'; char* str=%c%s%c; FILE *fd=fopen(%cGrace_kid.c%c,%cw%c); if (fd) { fprintf(fd, str, eol, eol, eol, eol, eol, eol, eol, eol, eol, esc, qt, esc, esc, qt, str, qt, qt, qt, qt, qt, qt, esc, qt, eol, eol, eol); FCLSE(fd); } else printf(%cerror%cn%c); return (x); }%c%cFT(zero)%c"; FILE *fd=fopen("Grace_kid.c","w"); if (fd) { fprintf(fd, str, eol, eol, eol, eol, eol, eol, eol, eol, eol, esc, qt, esc, esc, qt, str, qt, qt, qt, qt, qt, qt, esc, qt, eol, eol, eol); FCLSE(fd); } else printf("error\n"); return (x); }

FT(zero)

File: dr_quine/Bonus/Grace.py
'''
J'aime les poneys
'''

import sys

def percent():
    return '%'

def retx(x):
    return x

def FT(x):
    eol = '\n'
    qt = '"'
    esc = '\\'
    tab = "    "
    str = "'''%cJ'aime les poneys%c'''%c%cimport sys%c%cdef percent():%c%sreturn '%c'%c%cdef retx(x):%c%sreturn x%c%cdef FT(x):%c%seol = '%cn'%c%sqt = '%c'%c%sesc = '%c%c'%c%stab = %c    %c%c%sstr = %c%s%c%c%ssys.stdout = open('Grace_kid.py', 'w')%c%sprint str %c (eol, eol, eol, eol, eol, eol, eol, tab, percent(), eol, eol, eol, tab, eol, eol, eol, tab, esc, eol, tab, qt, eol, tab, esc, esc, eol, tab, qt, qt, eol, tab, qt, str, qt, eol, tab, eol, tab, percent(), eol, tab, eol, eol)%c%sretx(x)%c%cFT(0)"
    sys.stdout = open('Grace_kid.py', 'w')
    print str % (eol, eol, eol, eol, eol, eol, eol, tab, percent(), eol, eol, eol, tab, eol, eol, eol, tab, esc, eol, tab, qt, eol, tab, esc, esc, eol, tab, qt, qt, eol, tab, qt, str, qt, eol, tab, eol, tab, percent(), eol, tab, eol, eol)
    retx(x)

FT(0)
